lib/bolt/config.rb

```diff
@@ -39,114 +39,320 @@ class Config
       'remote' => Bolt::Config::Transport::Remote
     }.freeze
 
-    # Options that configure Bolt. These options are used in bolt.yaml and
-    # bolt-defaults.yaml.
-    BOLT_CONFIG = {
-      "color"               => "Whether to use colored output when printing messages to the console.",
-      "compile-concurrency" => "The maximum number of simultaneous manifest block compiles.",
-      "concurrency"         => "The number of threads to use when executing on remote targets.",
-      "format"              => "The format to use when printing results. Options are `human` and `json`.",
-      "plugin_hooks"        => "Which plugins a specific hook should use.",
-      "plugins"             => "A map of plugins and their configuration data.",
-      "puppetdb"            => "A map containing options for configuring the Bolt PuppetDB client.",
-      "puppetfile"          => "A map containing options for the `bolt puppetfile install` command.",
-      "save-rerun"          => "Whether to update `.rerun.json` in the Bolt project directory. If "\
-                               "your target names include passwords, set this value to `false` to avoid "\
-                               "writing passwords to disk."
-    }.freeze
-
-    # These options are only available to bolt-defaults.yaml.
-    DEFAULTS_CONFIG = {
-      "inventory-config" => "A map of default configuration options for the inventory. This includes options "\
-                            "for setting the default transport to use when connecting to targets, as well as "\
-                            "options for configuring the default behavior of each transport."
+    # The following constants define the various configuration options available to Bolt.
+    # Each constant is a hash where keys are the configuration option and values are the
+    # data describing the option. Data includes the following keys:
+    #   :def     The **documented** default value. This is the value that is displayed
+    #            in the reference docs and is not used by Bolt to actually set a default
+    #            value.
+    #   :desc    The text description of the option that is displayed in documentation.
+    #   :exmp    An example value for the option. This is used to generate an example
+    #            configuration file in the reference docs.
+    #   :type    The option's expected type. If an option accepts multiple types, this is
+    #            an array of the accepted types. Any options that accept a Boolean value
+    #            should use the TrueClass type.
+    OPTIONS = {
```

Comment:
This is awesome! The examples part is especially great. Organizationally, it might be nice if it was in a `Bolt::Config::Options` module so that it didn't take up so much real estate in the config class? Similar to the `Bolt::Shell::Powershell::Snippets` module.

Reply:
👍 I like that idea.